install/kubernetes/cilium/templates/spire/server/statefulset.yaml

```diff
@@ -36,14 +36,20 @@ spec:
       {{- end }}
       containers:
       - name: cilium-init
-        image: docker.io/library/busybox:1.35.0@sha256:223ae047b1065bd069aac01ae3ac8088b3ca4a527827e283b85112f29385fb1b
+        image: {{ include "cilium.image" .Values.authentication.mutual.spire.install.initImage | quote }}
+        imagePullPolicy: {{ .Values.authentication.mutual.spire.install.initImage.pullPolicy }}
         command:
           - /bin/sh
           - -c
           - |
             {{- tpl (.Files.Get "files/spire/init.bash") . | nindent 12 }}
       - name: spire-server
+        {{- if eq (typeOf .Values.authentication.mutual.spire.install.server.image) "string" }}
         image: {{ .Values.authentication.mutual.spire.install.server.image }}
+        {{- else }}
+        image: {{ include "cilium.image" .Values.authentication.mutual.spire.install.server.image | quote }}
```

Comment:
What does the "cilium.image" include bit do here exactly?

Reply:
Calling / Including the Helm template [cilium.image](https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/templates/_helpers.tpl#L8-L27). 

> Render full image name from given values, e.g:
> ```
> image:
>   repository: quay.io/cilium/cilium
>   tag: v1.10.1
>   useDigest: true
>   digest: abcdefgh
> ```
> then `include "cilium.image" .Values.image`
> will return `quay.io/cilium/cilium:v1.10.1@abcdefgh`

Reply:
I'm a bit confused by that. Why would someone use the Cilium image for the SPIRE server statefulset?

Reply:
I guess your point is why it is not 'spire.image' or 'xxx.image' ? 

actually, the `cilium.image` is used for all images with following statement, it mainly care about the structure, but does not care about what project the image belongs to. I think it means how to generate an image string by cilium manner, but does not mean generating cilium's image. Currently, it is also used to render etcd and envoy image which is not belonged to cilium.

```
{{/*
Render full image name from given values, e.g:
```
image:
  repository: quay.io/cilium/cilium
  tag: v1.10.1
  useDigest: true
  digest: abcdefgh
```
then `include "cilium.image" .Values.image`
will return `quay.io/cilium/cilium:v1.10.1@abcdefgh`
*/}}
{{- define "cilium.image" -}}
{{- $digest := (.useDigest | default false) | ternary (printf "@%s" .digest) "" -}}
{{- if .override -}}
{{- printf "%s" .override -}}
{{- else -}}
{{- printf "%s:%s%s" .repository .tag $digest -}}
{{- end -}}
{{- end -}}
```

It is also ok to define another declaration 'xxx.image' with same function code, or change the 'cilium.image' to be a common name

Reply:
Oh I see. `cilium.image` at face value sounds like an actual string instance of the Cilium image and not the template. We can probably improve the naming here to make it clear that this is an image template evaluation statement, but that's out-of-scope for this PR.

Reply:
yes, I would like to help rename `cilium.image` to be not misunderstood after the PR